src/main/java/org/jenkinsci/plugins/plaincredentials/impl/StringCredentialsImpl.java

```diff
@@ -39,7 +39,12 @@
 
     private final @Nonnull Secret secret;
 
-    @DataBoundConstructor public StringCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, @CheckForNull String description, @Nonnull Secret secret) {
+    @DataBoundConstructor public StringCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, @CheckForNull String description, @Nonnull String secret) {
+        super(scope, id, description);
+        this.secret = Secret.fromString(secret);
+    }
+
+    public StringCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, @CheckForNull String description, @Nonnull Secret secret) {
```

Comment:
```suggestion
    @Deprecated
    public StringCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, @CheckForNull String description, @Nonnull String secret) {
        this(scope, id, description, Secret.fromString(secret));
    }

    @DataBoundConstructor public StringCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, @CheckForNull String description, @Nonnull Secret secret) {
```

is what you want I think.

Reply:
The issue is that when set up the below DSL:
```
folder('my-tester') {
    properties {
        folderCredentialsProperty {
            domainCredentials {
                domainCredentials {
                    domain {
                        name("test")
                        description("Credentials necessary for our tests")
                    }
                    credentials {
                        stringCredentialsImpl {
                            scope("GLOBAL")
                            id("id")
                            description("")
                            secret("some string secret")
                        }
                    }
                }
            }
        }
    }
}
```
you get the following error:
```
ERROR: (script, line 15) No signature of method: javaposse.jobdsl.plugin.structs.DescribableContext.secret() is applicable for argument types: (java.lang.String) values: [some string secret]
Possible solutions: every(), getAt(java.lang.String), grep(), every(groovy.lang.Closure), split(groovy.lang.Closure), grep(java.lang.Object)
```

I presume (probably naively) that this is due to job-dsl being unable create a `Secret` object to pass it in. Changing it to be a string, makes above DSL work. From what I tested, Once I changed the constructor to accept a string instead of `Secret` object, job-dsl plugin correctly creates the credentials.

Reply:
Sounds like a bug/limitation in `job-dsl`.